### PR TITLE
2rp converge (pci cap, new deployment structure)

### PR DESCRIPTION
**Pull request by houlz0507**

### CODE REVIEW

**Comment by NaAl-Xlnx** [comment]
retest this please

File: src/runtime_src/tools/xclbin/SectionPartitionMetadata.h [hunk 1]
```diff
@@ -23,6 +23,7 @@
 #include "Section.h"
 #include "FDTProperty.h"
 #include <boost/functional/factory.hpp>
+#include <boost/property_tree/json_parser.hpp>
```
Comment (rozumx) [inline src/runtime_src/tools/xclbin/SectionPartitionMetadata.h]: Including headers from other headers should be kept to a minimum.   I don't believe that this jason_parser.hpp header is needed.

File: src/runtime_src/tools/xclbin/FDTNode.cxx [hunk 2]
```diff
@@ -178,10 +178,17 @@ FDTNode::FDTNode(const boost::property_tree::ptree& _ptDTC,
       XUtil::TRACE(XUtil::format("Property Found: %s", keyName.c_str()).c_str());
       FDTProperty * pProperty = new FDTProperty(iter, _propertyNameFormat);
       m_properties.push_back(pProperty);
-    } else {
+    } else if (!iter->second.empty()){
       XUtil::TRACE(XUtil::format("Node Found: %s", keyName.c_str()).c_str());
       FDTNode * pFDTNode = new FDTNode(iter->second, keyName, _propertyNameFormat);
       m_nestedNodes.push_back(pFDTNode);
+    } else {
```
Comment (rozumx) [inline src/runtime_src/tools/xclbin/FDTNode.cxx]: If I read this code correctly, you are desiring for all unknown variables to be strings.   For this schema, it is important to follow what is being requested by the Platform team.   Instead, variables should be defined in FDTProperty::PropertyNameFormat vector.  Please do so and remove the changes in this file.

File: src/runtime_src/tools/xclbin/SectionPartitionMetadata.cxx [hunk 3]
```diff
@@ -364,6 +364,18 @@ SchemaTransformToDTC_addressable_endpoints( const boost::property_tree::ptree& _
...
+SchemaTransformToDTC_partition_info ( const boost::property_tree::ptree& _ptOriginal,
+                                           boost::property_tree::ptree & _ptTransformed)
+{
+  std::ostringstream buf;
+  boost::property_tree::ptree ptInfo;
+
+  boost::property_tree::write_json(buf, _ptOriginal, false);
+  ptInfo.put("", buf.str());
+
+  _ptTransformed.add_child("__INFO", ptInfo);
```
Comment (rozumx) [inline src/runtime_src/tools/xclbin/SectionPartitionMetadata.cxx]: Hmm, "__INFO" looks out-of-place compared to the other schema variables....Is this really the case?

**Comment by rozumx** [comment]
Lizhi and I chatted and I understand what he is asking for.  I have updated the code review comments accordingly.

File: src/runtime_src/tools/xclbin/SectionPartitionMetadata.h (hunk 1 above)
Comment (houlz0507) [reply src/runtime_src/tools/xclbin/SectionPartitionMetadata.h]: this header is required for write_json(). Moved the include from .h to .cxx

File: src/runtime_src/tools/xclbin/FDTNode.cxx (hunk 2 above)
Comment (houlz0507) [reply src/runtime_src/tools/xclbin/FDTNode.cxx]: removed

File: src/runtime_src/tools/xclbin/SectionPartitionMetadata.cxx (hunk 3 above)
Comment (houlz0507) [reply src/runtime_src/tools/xclbin/SectionPartitionMetadata.cxx]: added { "__INFO", FDTProperty::DF_sz }

**Pull request merged**